kani-compiler/src/codegen_cprover_gotoc/overrides/hooks.rs

```diff
@@ -317,6 +317,72 @@ impl<'tcx> GotocHook<'tcx> for SliceFromRawPart {
     }
 }
 
+/// This hook intercepts calls to `memcmp` and skips CBMC's pointer checks if the number of bytes to be compared is zero.
+/// See issue <https://github.com/model-checking/kani/issues/1489>
+///
+/// This compiles `memcmp(first, second, count)` to:
+/// ```c
+/// count_var = count;
+/// first_var = first;
+/// second_var = second;
+/// count_var == 0 && first_var != NULL && second_var != NULL ? 0 : memcmp(first_var, second_var, count_var)
+/// ```
+pub struct MemCmp;
+
+impl<'tcx> GotocHook<'tcx> for MemCmp {
+    fn hook_applies(&self, tcx: TyCtxt<'tcx>, instance: Instance<'tcx>) -> bool {
+        let name = with_no_trimmed_paths!(tcx.def_path_str(instance.def_id()));
+        name == "core::slice::cmp::memcmp"
```

Comment:
Given the outdated discussion, can you also compare it against `"std::slice::cmp::memcmp"`.

Reply:
Done